lib/pier_logging/logger.rb

```diff
@@ -12,5 +12,16 @@ def initialize(*args)
     def create_formatter
       PierLogging.logger_configuration.formatter
     end
+
+    def _log(severity, *args)
+      redacted_args = redact_data(args)
+      super(severity, *redacted_args)
+    end
```

Comment:
Para que usamos isso?

Reply:
@banduk A gente usa pra obfuscar logs que não são do `PierLogging::RequestLogger`.

Internamente a gente usa o `Rails.logger` como o `PierLogging::Logger`, que por sua vez é classe filha do `Ougai::Logger`. 

O `Ougai::Logger` é uma classe que dá um `include Ougai::Logging`, que é um módulo com os métodos `debug`, `info`, `warn`, `error`, etc. São esses métodos que são chamados quando usamos um `Rails.logger.debug`, por exemplo.

Esse método aqui é pra dar um _override_ no `_log` do `Ougai::Logging`. Esse `_log` é usado por esses métodos acima:
https://github.com/tilfin/ougai/blob/main/lib/ougai/logging.rb#L41:L43

Então o que ta sendo feito, é que estamos capturando os dados que vão ser logados e obfuscando o que for necessário.